### PR TITLE
[PLAY-1647] Remove react modal from drawer 

### PR DESCRIPTION
**Pull request by markdoeswork**
**What does this PR do?** A clear and concise description with your runway ticket url.
Runway https://runway.powerhrg.com/backlog_items/PLAY-1647

Background: The drawer kit is currently in draft mode. Its similar to the dialog kit, and i used a lot of the same code to make the drawer kit.  

This story removes `react-modal` from the drawer kit. It didn't do as much as i thought it did and we can get away with just using a `div`

**How to test?** Steps to confirm the desired behavior:
1. Go to https://pr3891.playbook.beta.px.powerapp.cloud/kits/drawer/react
2. click all the examples

### CODE REVIEW

**Review by kangaree — APPROVED** [review]
Good work!

Next time, could you separate the formatting changes in a different commit? 
It was kind of hard to figure out what changes were related to the story.

File: playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx [hunk 1]
```diff
@@ -120,39 +129,54 @@ const Drawer = (props: DrawerProps): React.ReactElement => {
...
+  const [animationState, setAnimationState] = useState("");
+
+  useEffect(() => {
+    if (modalIsOpened) {
+      setAnimationState("afterOpen");
+    } else if (!modalIsOpened && animationState === "afterOpen") {
+      setAnimationState("beforeClose");
+      setTimeout(() => {
+        setAnimationState("");
+      }, 200); // closeTimeoutMS
```
Comment (kangaree) [inline playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx:153]: There's a bit of stuttering that shows up. I wonder if this should be tweaked?

https://github.com/user-attachments/assets/9b0df102-c95f-404b-bab8-79c31656f737

**Review by kangaree — CHANGES_REQUESTED** [review]

File: playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx [hunk 2]
```diff
@@ -120,39 +129,54 @@ const Drawer = (props: DrawerProps): React.ReactElement => {
...
+    if (modalIsOpened && behavior === "push" && body) {
+      if (placement === "left") {
         body.style.cssText = `margin-left: ${sizeMap[size]} !important; margin-right: '' !important;`;
-      } else if (placement === 'right') {
+      } else if (placement === "right") {
         body.style.cssText = `margin-right: ${sizeMap[size]} !important; margin-left: '' !important;`;
       }
 
-      body.classList.add('ReactModal__Body--open');
+      body.classList.add("ReactModal__Body--open");
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx:176]: Is there a reason, we are still tapping into the classes that get shipped from the ReactModal library? We shouldn't be needing these anymore.

File: playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx (hunk 1 above)
Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx:153]: Yeah I'd say, we either need to fix this animation in this current story, or get a fast-follow story for this to fix it. The kit is still in beta, but I'd rather fix it in this current story.

File: playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx (hunk 2 above)
Comment (markdoeswork) [reply playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx]: i changed the name to PBDrawer__Body--open

File: playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx (hunk 1 above)
Comment (markdoeswork) [reply playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx:153]: i fixed it check it out https://pr3891.playbook.beta.px.powerapp.cloud/kits/drawer/react

Comment (kangaree) [reply playbook/app/pb_kits/playbook/pb_drawer/_drawer.tsx:153]: Nice, looks a lot smoother on my end (using Firefox)!

**Review by jasperfurniss — APPROVED** [review]

**Review by kangaree — APPROVED** [review]

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**